Content.Client/Parallax/Managers/ParallaxManager.cs

```diff
@@ -15,91 +18,74 @@
-                return;
-
-            var parallaxConfig = GetParallaxConfig();
-            if (parallaxConfig == null)
-                return;
+    public async void LoadParallax()
+    {
+        await LoadParallaxByName("default");
+    }
 
-            var debugParallax = _configurationManager.GetCVar(CCVars.ParallaxDebug);
+    private async Task LoadParallaxByName(string name)
+    {
+        _parallaxName = name;
+        Logger.InfoS("parallax", $"Loading parallax {name}");
 
-            if (debugParallax
-                || !_resourceCache.UserData.TryReadAllText(PreviousParallaxConfigPath, out var previousParallaxConfig)
-                || previousParallaxConfig != parallaxConfig)
-            {
-                var table = Toml.ReadString(parallaxConfig);
-                await UpdateCachedTexture(table, debugParallax);
+        var parallaxPrototype = _prototypeManager.Index<ParallaxPrototype>(name);
 
-                //Update the previous config
-                using var writer = _resourceCache.UserData.OpenWriteText(PreviousParallaxConfigPath);
-                writer.Write(parallaxConfig);
-            }
+        var hq = await LoadParallaxLayers(parallaxPrototype.Layers.ToArray());
+        var lq = parallaxPrototype.LayersLQUseHQ ? hq : await LoadParallaxLayers(parallaxPrototype.LayersLQ.ToArray());
 
-            ParallaxTexture = GetCachedTexture();
-            OnTextureLoaded?.Invoke(ParallaxTexture);
+        if (_parallaxName == name)
+        {
+            _parallaxLayersHQ = hq;
+            _parallaxLayersLQ = lq;
+            Logger.InfoS("parallax", $"Loaded parallax {name}");
         }
-
-        private async Task UpdateCachedTexture(TomlTable config, bool saveDebugLayers)
+        else
         {
-            var debugImages = saveDebugLayers ? new List<Image<Rgba32>>() : null;
-
-            var sawmill = _logManager.GetSawmill("parallax");
-            // Generate the parallax in the thread pool.
-            using var newParallexImage = await Task.Run(() =>
-                ParallaxGenerator.GenerateParallax(config, new Size(1920, 1080), sawmill, debugImages));
-            // And load it in the main thread for safety reasons.
-
-            // Store it and CRC so further game starts don't need to regenerate it.
-            using var imageStream = _resourceCache.UserData.OpenWrite(ParallaxCachedImagePath);
-            newParallexImage.SaveAsPng(imageStream);
-
-            if (saveDebugLayers)
-            {
-                for (var i = 0; i < debugImages!.Count; i++)
-                {
-                    var debugImage = debugImages[i];
-                    using var debugImageStream = _resourceCache.UserData.OpenWrite(new ResourcePath($"/parallax_debug_{i}.png"));
-                    debugImage.SaveAsPng(debugImageStream);
-                }
-            }
+            Logger.InfoS("parallax", $"Loaded parallax {name}, but the target changed while it was being loaded.");
```

Comment:
Ok, done. Note that I'm not passing this to the PNG save functions because the parallax state can get a little weird if the PNG and the config get out of sync in any way.